packages/storage/Readme.md

```diff
@@ -1,5 +1,286 @@
+An example use-case for such a structure is if you want to contain a large amount of votes, deposits, or bets and iterate over them at some time in the future.
+Since iterating over large amounts of data at once may be prohibitive, this structure allows you to specify the amount of data that will
+be returned in each page.
+
+#### **Init**
+
+To import and intialize this storage object as a constant in `state.rs`, do the following:
+
+```rust
+use secret_toolkit_storage::{Keymap}
+```
+
+```rust
+pub const ADDR_VOTE: Keymap<HumanAddr, Foo> = Keymap::new(b"vote");
+pub const BET_STORE: Keymap<u32, BetInfo> = Keymap::new(b"vote");
+```
+
+You can use Json serde by doing `Keymap<HumanAddr, Uint128, Json>` similar to all the other storage objects above. However, keep in mind that the Serde algorthm is user to serde both the stored object (`Uint128`) AND the key (`HumanAddr`).
+
+If you need to associate a keymap to a user address (or any other variable), then you can also do this using the `.add_suffix` method.
+
+For example suppose that in your contract, a user can make multiple bets. Then, you'd want a Keymap to be associated to each user. You would achieve this my doing the following during execution in `contract.rs`.
+
+```rust
+// The compiler knows that user_bet_store is AppendStore<u32, BetInfo>
+let user_count_store = BET_STORE.add_suffix(env.message.sender.to_string().as_bytes());
+```
+
+#### **Read/Write**
+
+You can find more examples of using keymaps in the unit tests of Keymap in `keymap.rs`.
+
+To insert, remove, read from the keymap, do the following:
+
+```rust
+let user_addr: HumanAddr = env.message.sender;
+
+let foo = Foo {
+    message: "string one".to_string(),
+    votes: 1111,
+};
+
+ADDR_VOTE.insert(&mut deps.storage, &user_addr, foo.clone())?;
+// Compiler knows that this is Foo
+let read_foo = ADDR_VOTE.get(&deps.storage, &user_addr).unwrap();
+assert_eq!(read_foo, foo1);
+ADDR_VOTE.remove(&mut deps.storage, &user_addr)?;
+assert_eq!(ADDR_VOTE.get_len(&deps.storage)?, 0);
+```
+
+#### **Iterator**
+
+There are two methods that create an iterator in Keymap. These are `.iter` and `.iter_keys`. `iter_keys` only iterates over the keys whereas `iter` iterates over (key, item) pairs. Needless to say, `.iter_keys` is more efficient as it does not attempt to read the item.
+
+Keymap also has two paging methods, these are `.paging` and `.paging_keys`. `paging_keys` only paginates keys whereas `iter` iterates over (key, item) pairs. Needless to say, `.iter_keys` is more efficient as it does not attempt to read the item.
+
+Here are some select examples from the unit tests:
+
+```rust
+#[test]
+    fn test_keymap_iter_keys() -> StdResult<()> {
+        let mut storage = MockStorage::new();
```

Comment:
can you remove the indentation on the function? it looks weird in the rendered file